vm_network_migration/modules/unmanaged_instance_group.py

```diff
@@ -0,0 +1,187 @@
+            instanceGroup=self.instance_group_name)
+        while request is not None:
+            response = request.execute()
+            # no instances in the instance group
+            if 'items' not in response:
+                return []
+            for instance_with_named_ports in response['items']:
+                print(instance_with_named_ports)
+                instance_name = \
+                    instance_with_named_ports['instance'].split('instances/')[1]
+                instances.append(
+                    Instance(self.compute, self.project, instance_name,
+                             self.region, self.zone))
+            request = self.compute.instanceGroups().listInstances_next(
+                previous_request=request, previous_response=response)
+        return instances
+
+    def delete_instance_group(self) -> dict:
+        """ Delete the instance group in the compute engine
+
+        Returns:  a deserialized object of the response
+
+        """
+
+        delete_instance_group_operation = self.compute.instanceGroups().delete(
+            project=self.project,
+            zone=self.zone,
+            instanceGroup=self.instance_group_name).execute()
+        self.operation.wait_for_zone_operation(
+            delete_instance_group_operation['name'])
+        return delete_instance_group_operation
+
+    def create_instance_group(self, configs) -> dict:
+        """ Create the instance group
+
+        Returns: a deserialized object of the response
+
+        """
+        create_instance_group_operation = self.compute.instanceGroups().insert(
+            project=self.project,
+            zone=self.zone,
+            body=configs).execute()
+        self.operation.wait_for_zone_operation(
+            create_instance_group_operation['name'])
+        if configs == self.original_instance_group_configs:
+            self.migrated = False
+        elif configs == self.new_instance_group_configs:
+            self.migrated = True
+        return create_instance_group_operation
+
+    def add_an_instance(self, instance_selfLink) -> dict:
+        """ Add an instance into the instance group
+
+        Args:
+            instance_selflink: the instance's selfLink
+
+        Returns: a deserialized object of the response
+        Raises: HttpError
+
+        """
+        try:
+            add_instance_operation = self.compute.instanceGroups().addInstances(
+                project=self.project,
+                zone=self.zone,
+                instanceGroup=self.instance_group_name,
+                body={
+                    "instances": [{
+                        "instance": instance_selfLink}]}).execute()
+            self.operation.wait_for_zone_operation(
+                add_instance_operation['name'])
+            return add_instance_operation
+        except HttpError as e:
+            error_reason = e._get_reason()
+            if 'already' in error_reason:
```

Comment:
nit. make the error string a bit longer to make it clearer

Reply:
Changed to 'already a member of'